tests/unittests/tests-gnrc_ipv6_nib/tests-gnrc_ipv6_nib-internal.c

```diff
@@ -0,0 +1,1096 @@
+}
+
+/*
+ * Tries to get the default router from a list of one unreachable routers
+ * Expected result: _nib_drl_get_dr() returns always the one unreachable router
+ */
+static void test_nib_drl_get_dr__round_robin1(void)
+{
+    _nib_dr_entry_t *node, *nib_res;
+    ipv6_addr_t addr = { .u64 = { { .u8 = GLOBAL_PREFIX },
+                                  { .u64 = TEST_UINT64 } } };
+
+    TEST_ASSERT_NOT_NULL((node = _nib_drl_add(&addr, IFACE)));
+    node->next_hop->info = GNRC_IPV6_NIB_NC_INFO_NUD_STATE_UNREACHABLE;
+    TEST_ASSERT_NOT_NULL((nib_res = _nib_drl_get_dr()));
+    TEST_ASSERT(nib_res == node);
+    TEST_ASSERT_NOT_NULL((nib_res = _nib_drl_get_dr()));
+    TEST_ASSERT(nib_res == node);
+    TEST_ASSERT_NOT_NULL((nib_res = _nib_drl_get_dr()));
+    TEST_ASSERT(nib_res == node);
+    TEST_ASSERT_NOT_NULL((nib_res = _nib_drl_get_dr()));
+    TEST_ASSERT(nib_res == node);
+}
+
+/*
+ * Tries to get the default router from a list of two unreachable routers
+ * Expected result: _nib_drl_get_dr() returns one router first, then the other,
+ * then the first again, etc.
+ */
+static void test_nib_drl_get_dr__round_robin2(void)
+{
+    _nib_dr_entry_t *node1, *node2, *nib_res;
+    ipv6_addr_t addr = { .u64 = { { .u8 = GLOBAL_PREFIX },
+                                  { .u64 = TEST_UINT64 } } };
+
+    TEST_ASSERT_NOT_NULL((node1 = _nib_drl_add(&addr, IFACE)));
+    addr.u64[1].u64++;
+    node1->next_hop->info = GNRC_IPV6_NIB_NC_INFO_NUD_STATE_UNREACHABLE;
+    TEST_ASSERT_NOT_NULL((node2 = _nib_drl_add(&addr, IFACE)));
+    node2->next_hop->info = GNRC_IPV6_NIB_NC_INFO_NUD_STATE_INCOMPLETE;
+    TEST_ASSERT_NOT_NULL((nib_res = _nib_drl_get_dr()));
+    TEST_ASSERT(nib_res == node1);
+    TEST_ASSERT_NOT_NULL((nib_res = _nib_drl_get_dr()));
+    TEST_ASSERT(nib_res == node2);
+    TEST_ASSERT_NOT_NULL((nib_res = _nib_drl_get_dr()));
+    TEST_ASSERT(nib_res == node1);
+    TEST_ASSERT_NOT_NULL((nib_res = _nib_drl_get_dr()));
+    TEST_ASSERT(nib_res == node2);
+}
+
+/*
+ * Tries to get the default router from a list of two routers
+ * the first reachable, the second unreachable
+ * Expected result: _nib_drl_get_dr() always returns the first router
+ */
+static void test_nib_drl_get_dr__success1(void)
+{
+    _nib_dr_entry_t *node1, *node2, *nib_res;
+    ipv6_addr_t addr = { .u64 = { { .u8 = GLOBAL_PREFIX },
+                                  { .u64 = TEST_UINT64 } } };
+
+    TEST_ASSERT_NOT_NULL((node1 = _nib_drl_add(&addr, IFACE)));
+    addr.u64[1].u64++;
+    node1->next_hop->info = GNRC_IPV6_NIB_NC_INFO_NUD_STATE_STALE;
+    TEST_ASSERT_NOT_NULL((node2 = _nib_drl_add(&addr, IFACE)));
+    node2->next_hop->info = GNRC_IPV6_NIB_NC_INFO_NUD_STATE_INCOMPLETE;
+    TEST_ASSERT_NOT_NULL((nib_res = _nib_drl_get_dr()));
+    TEST_ASSERT(nib_res == node1);
+    TEST_ASSERT_NOT_NULL((nib_res = _nib_drl_get_dr()));
+    TEST_ASSERT(nib_res == node1);
+}
+
+/*
+ * Tries to get the default router from a list of two routers
```

Comment:
tests with two routers are obsolete when you are testing the permutation of three routers below ..